packages/select/src/components/query-list/queryList.tsx

```diff
@@ -350,6 +369,40 @@ export class QueryList<T> extends React.Component<IQueryListProps<T>, IQueryList
         }
     };
 
+    private handlePaste = (values: string[]) => {
+        const { createNewItemFromQuery, onItemsPaste } = this.props;
+
+        let nextActiveItem: T | undefined;
+
+        // Find an exising item that exactly matches each pasted value, or
+        // create a new item if possible. Ignore unmatched values if creating
+        // items is disabled.
+        const pastedItemsToEmit = values.reduce<T[]>((agg, value) => {
```

Comment:
nit: I would slightly prefer this as a regular for loop, since that would be easier to read than a `reduce` which produces a list